### ISSUE TITLE
jax.numpy.sqrt ignores negative zero imaginary part on complex branch cut

### ISSUE THREAD

**Issue by ALinrunrun** [jax-ml/jax#39110]
### Description

`jax.numpy.sqrt` appears to return the wrong branch for complex inputs on the negative real axis when the imaginary part is `-0.0`.

For complex square root, the sign of zero in the imaginary component determines which side of the branch cut the input approaches from. NumPy preserves this behavior: `sqrt(-1.0 - 0.0j)` returns `-1j`, while `sqrt(-1.0 + 0.0j)` returns `+1j`.

However, JAX returns the upper-branch result for inputs with imaginary part `-0.0`.

### Minimal reproducible example

```
#!/usr/bin/env python3
import os
import sys
import warnings

warnings.filterwarnings("ignore")
os.environ.setdefault("JAX_PLATFORMS", "cpu")

try:
    import numpy as np
    import jax.numpy as jnp
except ImportError as e:
    print(f"missing dep: {e}")
    sys.exit(2)

# Negative real axis approached from below:
# imag = -0.0 should select the lower branch.
x = np.array(
    [complex(-1.0, -0.0), complex(-2.0, -0.0)],
    dtype=np.complex64,
)

out = np.asarray(jnp.sqrt(jnp.asarray(x)))
expected = np.sqrt(x.astype(np.complex128)).astype(np.complex64)

print("jax:", out.tolist())
print("expected:", expected.tolist())

if not np.allclose(out, expected, atol=1e-4, rtol=1e-3, equal_nan=True):
    sys.exit(0)
sys.exit(1)
```

### Actual result

```
jax: [1j, 1.4142135381698608j]
expected: [-1j, -1.4142135381698608j]
```

### Expected result

```
[-1j, -1.4142135381698608j]
```

The inputs are on the negative real axis with imaginary part `-0.0`, so the complex square root should follow the lower branch. JAX appears to lose or ignore the sign of the zero imaginary component and returns the upper-branch result instead.

### System info (python version, jaxlib version, accelerator, etc.)

jax: 0.10.1
jaxlib: 0.10.1
numpy: 2.4.6
python: 3.13.13
platform: Ubuntu 24.04.3 LTS, Linux x86_64
accelerator: CPU backend

Note: GPU is present, but CUDA-enabled jaxlib is not installed, so JAX falls back to CPU.

**Comment by jakevdp** [comment]
This falls under https://github.com/jax-ml/jax/issues/38651; please note that it may take a while for this to receive attention from the team.

**Comment by CodersAcademy006** [comment]
Reproduced on 0.10.2: `jnp.sqrt(complex(-1.0, -0.0))` returns `1j` where NumPy returns `-1j`. The reason is that `jnp.sqrt` (`jax/_src/numpy/ufuncs.py:1165`) is a thin wrapper that defers entirely to `lax.sqrt`, so the branch selection on the negative real axis happens in XLA's complex-sqrt lowering, which is not distinguishing `-0.0` from `+0.0` in the imaginary component. NumPy follows the C99/Annex-G convention where the sign of the zero imaginary part picks the side of the branch cut (`sqrt(-1-0j) = -1j`, `sqrt(-1+0j) = +1j`), so this looks like the lowering collapsing signed zero before the branch decision rather than anything in the Python layer. Fixing it likely needs either the XLA complex-sqrt path to preserve the imaginary sign, or a jax-level complex special-case that routes the negative-real, negative-zero-imaginary input to the lower branch. @jakevdp
